SwiftUIStarter/SwiftUIStarter/NativeDisplayView.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  NativeDisplayUIView.swift
+//  SwiftUIStarter
+//
+//  Copyright © 2025 CleverTap. All rights reserved.
+//
+
+import SwiftUI
+import CleverTapSDK
+
+struct NativeDisplayUIView: View {
+    @StateObject private var viewModel = NativeDisplayUIViewModel()
+    @State private var displayUnitID = ""
+    @State private var showingAlert = false
+    @State private var alertMessage = ""
+    @State private var alertTitle = ""
+    
+    var body: some View {
+        NavigationView {
+            VStack(spacing: 0) {
+                VStack(spacing: 16) {
+                    HStack {
+                        Text("NATIVE DISPLAY\nNOTE: CLICKING ON BELOW BUTTON WILL RECORD EVENT WITH SAME NAME.")
+                            .font(.caption)
+                            .fontWeight(.medium)
+                            .foregroundColor(.gray)
+                        Spacer()
+                    }
+                    .padding(.horizontal, 16)
+                    .padding(.top, 16)
+                    
+                    // Display type buttons
+                    HStack(spacing: 20) {
+                        Button("ND_Simple") {
+                            viewModel.handleDisplayType(.simple)
+                        }
+                        .foregroundColor(.blue)
+                        .font(.system(size: 16))
+                        
+                        Button("ND_Carousel") {
+                            viewModel.handleDisplayType(.carousel)
+                        }
+                        .foregroundColor(.blue)
+                        .font(.system(size: 16))
+                    }
+                    .padding(.horizontal, 16)
+                    
+                    // Display unit ID input and Get Unit button
+                    HStack(spacing: 12) {
+                        TextField("Display unit ID", text: $displayUnitID)
+                            .textFieldStyle(RoundedBorderTextFieldStyle())
+                        
+                        Button("Get Unit") {
+                            viewModel.getDisplayUnit(withID: displayUnitID)
```

Comment:
This replaces the displayed units with the one with the id (if found). I would recommend updating the Display Units list Header depending on the items shown, i.e., All Display Units or Display Unit with ID.